src/js/components/TabPanesComponent.jsx

```diff
@@ -33,6 +33,36 @@ var TabPanesComponent = React.createClass({
     this.updateCurrentGroup();
   },
 
+  getContextualBar: function () {
+    var state = this.state;
+
+    if (state.filters.filterText == null ||
+        Util.isEmptyString(state.filters.filterText)) {
+      return <BreadcrumbComponent groupId={state.currentGroup} />;
+    }
+
+    return (
+        <p className="breadcrumb">
+          <span>{`Search results for "${state.filters.filterText}"`}</span>
+          <a href="#"
+              className="clear"
+              onClick={this.handleClearFilterText}>
+            Clear search
+          </a>
+        </p>
+    );
+  },
+
+  handleClearFilterText: function (event) {
```

Comment:
OK, I got wrong, in SidebarComponent we introduced ```getClearLinkForFilter()```, this could go into a mixin.